### PR TITLE
RJS-2913: Supporting React Native v0.76.0

### LINKED ISSUE #6910: Upgrade React Native Test App to RN 0.76.0

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This upgrades our test app to use React Native 0.76.0 and closes #6910 and #6908.

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [x] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary
* [ ] 🔔 Mention `@realm/devdocs` if documentation changes are needed

### CODE REVIEW

File: packages/realm-react/src/__tests__/AppProvider.test.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ import { act, fireEvent, render, renderHook, waitFor } from "@testing-library/re
 import { AppProvider, useApp } from "../AppProvider";
 import { Realm } from "realm";
 
-jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
+// jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
```
Comment (gagik) [inline packages/realm-react/src/__tests__/AppProvider.test.tsx:26]: Should it be left as a comment or just removed?

**Review by gagik — APPROVED** [review]
Not too familiar with the intricacies here but looks reasonable, just one note about the commented code.

File: packages/realm-react/src/__tests__/AppProvider.test.tsx (hunk 1 above)
Comment (kraenhansen) [reply packages/realm-react/src/__tests__/AppProvider.test.tsx:26]: To be honest, I don't understand the original need for this line and the file is missing in the latest version of React Native, leading to a test failure.

I left it as a comment to make it easier to search for and blame if it starts failing 🙈

Comment (gagik) [reply packages/realm-react/src/__tests__/AppProvider.test.tsx:26]: Fair enough, although maybe it'd be good to have a short comment just stating that

**Pull request merged**